### DISCUSSION TITLE
ct/h vs €/h

### DISCUSSION THREAD

**Discussion by kobius77** [mampfes/ha_epex_spot#122] [Announcements]
not sure if i'm missing something - but HA's Energy Dashboard is expecting prices as €/h but this integration is providing ct/h in the 'Net Price' sensor .. so i just was a little shocked to see last month's cost of energy ;)

of course i could create a template sensor - or is there a way to change the units for the dashboard?

but wouldn't it make sense to align the units to make things easier?

~~(sorry, didn't see this repo had discussions enabled)~~

**Comment by mampfes** [comment]
Hi, this is on my list, but I didn't had time to work on it.

My solution would be to switch all prices to EUR/kWh (or GB/kWh respectively). This would simplify a lot of the price handling today. But it also means that if s.o. wants to see price in ct/kWh, they have to add a template sensor or reconfigure their charts.

**Reply by kobius77** [reply]
that's an obvious worry. but since this integration is for HA and (i think) most things in HA are using CurrencyUnit/kWh for energy cost it probably would make more sense to align with that as there'll likely be more users to come in future than are using it by now ;)

the other option of course would be to add an additional sensor, though a self explanatory name probably would be a challenge to find.. "Net100 Price"?

**Comment by CommanderRedYT** [comment]
I would really like to bump this! Unfortunately I am not that familiar with Homeassistant, but I could still try and create a PR

**Comment by suaveolent** [comment]
Thanks for this awesome integration. I would also like to bump this topic. Using €/kWh seems to be the preferred way for Home Assistant.

**Comment by CommanderRedYT** [comment]
As a temporary solution I created a template sensor with the following config:

template-string:
```python
{{ float(states('sensor.epex_spot_data_net_price')) / 100 | default(0) }}
```

Unit of measurement: EUR/kWh

everything else empty. works fine now since days and price seems to be correct. this is what i then use in the energy dashboard

**Comment by suaveolent** [comment]
Thanks, I have just created a PR which changes the default unit to € / kWh (see https://github.com/mampfes/ha_epex_spot/pull/128)

**Comment by suaveolent** [comment]
@drahdiwaberl @CommanderRedYT @ericdevos I have finally merged the PR and released a pre-release. You are more than welcome to test, to see if there are any issues with the new implementation.

**Reply by aichingerk** [reply]
@suaveolent 
I test your new implementation. The only thing i found was for the configuration for the absolute surcharge. it is not possible to insert 0.012. (I mean with the keyboard the dot) Only copy past is working for the surcharge.

For the Energy Dashboard, it's nice, that now i can use the net price sensor itself.

For the Apex chart, i use in the y scale ct/kWh it is nicer to see. The changes are:
```
    data_generator: >
      return entity.attributes.data.map((entry, index) => { return [new
      Date(entry.start_time).getTime(), entry.price_per_kwh * 100]; });  
```

For my automation's i used also ct prices and so i changed my templates like:
```
sensor:
  - name: epex_high_intervall_afternoon_2h_price
    unique_id: epex_high_intervall_afternoon_2h_price
    state: "{% set price = (resp.net_price_per_kwh is defined and resp.net_price_per_kwh) %}
            {{ '{:.4f}'.format(price * 100) }}"
```

nice work, Thx

**Reply by suaveolent** [reply]
Thanks for your feedback.

I noticed that issue too. Might be an issue which needs to be reported to Home Assistant Core. 

I was also thinking of adding the ct/kWh value to the attributes. So no conversion would be necessary any more.

**Comment by TobHaeg** [comment]
I was also irritated about the long numbers in the €/kwh price. so i opened days ago an issue https://github.com/mampfes/ha_epex_spot/issues/210
